There is one remaining CMake build issue in the CI:

```
[ 80%] Building C object syslog-ng-ctl/CMakeFiles/syslog-ng-ctl.dir/syslog-ng-ctl.c.o
/__w/syslog-ng/syslog-ng/modules/file-rotation/file-rotation.c:25:10: fatal error: 'modules/affile/file-signals.h' file not found
#include "modules/affile/file-signals.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

You can fix this by adding the following to `modules/file-rotation/CMakeLists.txt`:

```diff
--- a/modules/file-rotation/CMakeLists.txt
+++ b/modules/file-rotation/CMakeLists.txt
@@ -15,6 +15,7 @@ add_module(
   TARGET file_rotation
   GRAMMAR file-rotation-grammar
   SOURCES ${FILE_ROTATION_SOURCES}
+  INCLUDES ${PROJECT_SOURCE_DIR}
 )
```

--- a/modules/affile/affile-dest.c
+++ b/modules/affile/affile-dest.c
@@ -163,7 +163,7 @@ affile_dw_reopen(AFFileDestWriter *self)
       LogTransport *transport = file_opener_construct_transport(self->owner->file_opener, fd);
 
       proto = file_opener_construct_dst_proto(self->owner->file_opener, transport,
-                                              &self->owner->writer_options.proto_options.super);
+                                              &self->owner->writer_options.proto_options.super, self->owner->super.super.super.signal_slot_connector);
     }
   else if (open_result == FILE_OPENER_RESULT_ERROR_PERMANENT)
     {

--- a/modules/affile/file-opener.h
+++ b/modules/affile/file-opener.h
@@ -28,6 +28,7 @@
 #include "compat/lfs.h"
 
 #include "file-perms.h"
+#include "signal-slot-connector/signal-slot-connector.h"
 #include "transport/logtransport.h"
 #include "logproto/logproto-server.h"
 #include "logproto/logproto-client.h"
@@ -64,7 +65,8 @@ struct _FileOpener
   LogTransport *(*construct_transport)(FileOpener *self, gint fd);
   LogProtoServer *(*construct_src_proto)(FileOpener *self, LogTransport *transport,
                                          LogProtoFileReaderOptions *proto_options);
-  LogProtoClient *(*construct_dst_proto)(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options);
+  LogProtoClient *(*construct_dst_proto)(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options,
+                                         SignalSlotConnector *connector);
 };
 
 static inline LogTransport *
@@ -80,9 +82,10 @@ file_opener_construct_src_proto(FileOpener *self, LogTransport *transport, LogPr
 }
 
 static inline LogProtoClient *
-file_opener_construct_dst_proto(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options)
+file_opener_construct_dst_proto(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options,
+                                SignalSlotConnector *connector)
 {
-  return self->construct_dst_proto(self, transport, proto_options);
+  return self->construct_dst_proto(self, transport, proto_options, connector);
 }
 
 FileOpenerResult file_opener_open_fd(FileOpener *self, const gchar *name, FileDirection dir, gint *fd);

--- a/modules/affile/file-signals.h
+++ b/modules/affile/file-signals.h
@@ -1,17 +1,18 @@
 /*
  * Copyright (c) 2022 Shikhar Vashistha
  *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of the GNU General Public License version 2 as published
- * by the Free Software Foundation, or (at your option) any later version.
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
  *
- * This program is distributed in the hope that it will be useful,
+ * This library is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
  *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  *
  * As an additional exemption you are allowed to compile & link against the
@@ -23,7 +24,14 @@
 #ifndef FILE_SIGNALS_H_INCLUDED
 #define FILE_SIGNALS_H_INCLUDED
 
+#include "signal-slot-connector/signal-slot-connector.h"
+
 typedef struct _FileFlushSignalData FileFlushSignalData;
+
+struct _FileFlushSignalData
+{
+  const gchar *filename;
+};
 
 #define signal_file_flush SIGNAL(file_rotation, rotation_request, FileFlushSignalData *)
 

--- a/modules/affile/logproto-file-writer.c
+++ b/modules/affile/logproto-file-writer.c
@@ -23,7 +23,6 @@
 
 #include "logproto-file-writer.h"
 #include "messages.h"
-#include "affile-dest.h"
 #include "file-signals.h"
 #include <string.h>
 #include <errno.h>
@@ -34,7 +33,6 @@ typedef struct _LogProtoFileWriter
 {
   LogProtoClient super;
   SignalSlotConnector *signal_slot_connector;
-  AFFileDestDriver *signal_slot_connector_driver;
   guchar *partial;
   gsize partial_len, partial_pos;
   gint partial_messages;
@@ -137,6 +135,10 @@ log_proto_file_writer_flush(LogProtoClient *s)
   self->buf_count = 0;
   self->sum_len = 0;
 
+  FileFlushSignalData signal_data = {};
+
+  EMIT(self->signal_slot_connector, signal_file_flush, &signal_data);
+
   return LPS_SUCCESS;
 
 write_error:
@@ -148,24 +150,6 @@ write_error:
                 evt_tag_error(EVT_TAG_OSERROR));
       return LPS_ERROR;
     }
-  // emit a "flush" signal to the transport, so that it can react to the error
-  GString *request_body = g_string_new("");
-  g_string_append_printf(request_body, "%d", self->buf_size);
-  GString *file_name = g_string_new("");
-  g_string_append_printf(file_name, "%s", self->super.transport->name);
-  FileFlushSignalData signal_data =
-  {
-    .filename = file_name,
-    .size = self->buf_size, // implementation yet to be decided
-    .interval = 0, // implementation yet to be decided
-  };
-  //pass the AFFileDestDriver's signal_slot_connector to LogProtoFileWriter's signal_slot_connector
-  LogProtoFileWriter *self_cast = (LogProtoFileWriter *)self;
-  SignalSlotConnector *signal_slot_connector = (SignalSlotConnector *)
-                                               self->signal_slot_connector_driver->super.super.super.signal_slot_connector;
-  self_cast->signal_slot_connector = signal_slot_connector;
-
-  EMIT(signal_slot_connector, signal_file_flush, &signal_data);
 
   return LPS_SUCCESS;
 }
@@ -238,7 +222,8 @@ log_proto_file_writer_prepare(LogProtoClient *s, gint *fd, GIOCondition *cond, g
 }
 
 LogProtoClient *
-log_proto_file_writer_new(LogTransport *transport, const LogProtoClientOptions *options, gint flush_lines, gint fsync_)
+log_proto_file_writer_new(LogTransport *transport, const LogProtoClientOptions *options, gint flush_lines, gint fsync_,
+                          SignalSlotConnector *connector)
 {
   if (flush_lines == 0)
     /* the flush-lines option has not been specified, use a default value */
@@ -253,8 +238,11 @@ log_proto_file_writer_new(LogTransport *transport, const LogProtoClientOptions *
   LogProtoFileWriter *self = (LogProtoFileWriter *)g_malloc0(sizeof(LogProtoFileWriter) + sizeof(
       struct iovec)*flush_lines);
 
+
+
   log_proto_client_init(&self->super, transport, options);
-  self->signal_slot_connector = signal_slot_connector_new();
+
+  self->signal_slot_connector = connector;
   self->fd = transport->fd;
   self->buf_size = flush_lines;
   self->fsync = fsync_;

--- a/modules/affile/logproto-file-writer.h
+++ b/modules/affile/logproto-file-writer.h
@@ -25,9 +25,9 @@
 #define LOG_PROTO_FILE_WRITER_H_INCLUDED
 
 #include "logproto/logproto-client.h"
-#include "modules/file-rotation/file-rotation.h"
+#include "signal-slot-connector/signal-slot-connector.h"
 
 LogProtoClient *log_proto_file_writer_new(LogTransport *transport, const LogProtoClientOptions *options,
-                                          gint flush_lines, gboolean fsync);
+                                          gint flush_lines, gboolean fsync, SignalSlotConnector *connector);
 
 #endif

--- a/modules/affile/named-pipe.c
+++ b/modules/affile/named-pipe.c
@@ -92,7 +92,8 @@ _construct_src_proto(FileOpener *s, LogTransport *transport, LogProtoFileReaderO
 }
 
 static LogProtoClient *
-_construct_dst_proto(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options)
+_construct_dst_proto(FileOpener *self, LogTransport *transport, LogProtoClientOptions *proto_options,
+                     SignalSlotConnector *connector)
 {
   return log_proto_text_client_new(transport, proto_options);
 }

--- a/modules/affile/regular-files.c
+++ b/modules/affile/regular-files.c
@@ -85,13 +85,14 @@ typedef struct _FileOpenerRegularDestFiles
 } FileOpenerRegularDestFiles;
 
 static LogProtoClient *
-_construct_dst_proto(FileOpener *s, LogTransport *transport, LogProtoClientOptions *proto_options)
+_construct_dst_proto(FileOpener *s, LogTransport *transport, LogProtoClientOptions *proto_options,
+                     SignalSlotConnector *connector)
 {
   FileOpenerRegularDestFiles *self = (FileOpenerRegularDestFiles *) s;
 
   return log_proto_file_writer_new(transport, proto_options,
                                    self->writer_options->flush_lines,
-                                   *self->use_fsync);
+                                   *self->use_fsync, connector);
 }
 
 static LogTransport *

--- a/modules/affile/tests/test_file_writer.c
+++ b/modules/affile/tests/test_file_writer.c
@@ -35,6 +35,7 @@ static void _ack_callback(gint num_acked, gpointer user_data);
 static LogProtoClientOptions options = {0};
 static LogMessage *msg;
 static LogTransport *transport;
+static SignalSlotConnector *connector;
 static LogProtoClientFlowControlFuncs flow_control_funcs =
 {
   .ack_callback = _ack_callback,
@@ -54,7 +55,8 @@ _ack_callback(gint num_acked, gpointer user_data)
 
 Test(file_writer, write_single_message_and_flush_is_expected_to_dump_the_payload_to_the_output)
 {
-  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, 100, FALSE);
+  connector = signal_slot_connector_new();
+  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, 100, FALSE, connector);
 
   log_proto_client_set_client_flow_control(fw, &flow_control_funcs);
 
@@ -76,7 +78,9 @@ Test(file_writer, batches_of_messages_should_be_flushed_to_the_output)
 {
   const gint BATCH_SIZE = 10;
   const gint MESSAGE_COUNT = BATCH_SIZE * 3;
-  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, BATCH_SIZE, FALSE);
+  connector = signal_slot_connector_new();
+
+  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, BATCH_SIZE, FALSE, connector);
 
   log_proto_client_set_client_flow_control(fw, &flow_control_funcs);
   for (gint i = 0; i < MESSAGE_COUNT; i++)
@@ -105,7 +109,9 @@ Test(file_writer, batches_of_messages_should_be_flushed_to_the_output)
 Test(file_writer, messages_should_be_flushed_automatically_once_we_reach_batch_size)
 {
   const gint BATCH_SIZE = 10;
-  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, BATCH_SIZE, FALSE);
+  connector = signal_slot_connector_new();
+
+  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, BATCH_SIZE, FALSE, connector);
 
   log_proto_client_set_client_flow_control(fw, &flow_control_funcs);
   for (gint i = 0; i < BATCH_SIZE - 1; i++)
@@ -133,7 +139,9 @@ Test(file_writer, messages_should_be_flushed_automatically_once_we_reach_batch_s
 Test(file_writer, batches_of_messages_are_flushed_even_if_the_underlying_transport_is_accepting_a_few_bytes_per_write)
 {
   const gint BATCH_SIZE = 10;
-  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, BATCH_SIZE, FALSE);
+  connector = signal_slot_connector_new();
+
+  LogProtoClient *fw = log_proto_file_writer_new(transport, &options, BATCH_SIZE, FALSE, connector);
 
   log_transport_mock_set_write_chunk_limit((LogTransportMock *) transport, 2);
   log_proto_client_set_client_flow_control(fw, &flow_control_funcs);

--- a/modules/file-rotation/CMakeLists.txt
+++ b/modules/file-rotation/CMakeLists.txt
@@ -14,7 +14,7 @@ set(FILE_ROTATION_SOURCES
 add_module(
   TARGET file_rotation
   GRAMMAR file-rotation-grammar
+  INCLUDES ${PROJECT_SOURCE_DIR}
   SOURCES ${FILE_ROTATION_SOURCES}
 )
-
 add_test_subdirectory(tests)

--- a/modules/file-rotation/file-rotation-grammar.ym
+++ b/modules/file-rotation/file-rotation-grammar.ym
@@ -81,7 +81,7 @@ file_rotation_options
 
 file_rotation_option
   : KW_SIZE '(' nonnegative_integer ')' { file_rotation_set_size(last_file_rotation_plugin, $3); }
-  | KW_INTERVAL '(' string ')' { log_rotation_set_interval(last_file_rotation_plugin, $3); free($3); }
+  | KW_INTERVAL '(' string ')' { file_rotation_set_interval(last_file_rotation_plugin, $3); free($3); }
   ;
 
 /* INCLUDE_RULES */

--- a/modules/file-rotation/file-rotation.c
+++ b/modules/file-rotation/file-rotation.c
@@ -22,15 +22,13 @@
  */
 
 #include "file-rotation.h"
-#include "modules/http/http-signals.h"
 #include "modules/affile/file-signals.h"
 #include "driver.h"
 
 static void
-_slot_append_file_rotation_request(FileRotationPlugin *self, FileFlushSignalData *data)
+_slot_file_rotation(FileRotationPlugin *self, FileFlushSignalData *data)
 {
-
-  g_string_append_printf(self->filename, "%s", data->filename->str);
+  msg_trace("File rotation signal received", evt_tag_str("filename", data->filename));
 }
 
 static gboolean
@@ -38,9 +36,9 @@ file_rotation_attach_to_driver(LogDriverPlugin *s, LogDriver *driver)
 {
   SignalSlotConnector *ssc = driver->super.signal_slot_connector;
 
-  msg_debug("File rotation plugin has been attached", evt_tag_str("to", s->name));
+  msg_debug("File rotation plugin has been attached", evt_tag_str("to", driver->id));
 
-  CONNECT(ssc, signal_file_flush, _slot_append_file_rotation_request, s);
+  CONNECT(ssc, signal_file_flush, _slot_file_rotation, s);
 
   return TRUE;
 }
@@ -50,17 +48,13 @@ file_rotation_free(LogDriverPlugin *s)
 {
   msg_debug("FileRotationPlugin::free");
   FileRotationPlugin *self = (FileRotationPlugin *)s;
-  g_free(self->interval);
+  g_free(self->filename);
   log_driver_plugin_free_method(s);
 }
 void
-log_rotation_set_interval(FileRotationPlugin *self, char *interval)
+file_rotation_set_interval(FileRotationPlugin *self, gchar *interval)
 {
-  FileRotationPlugin *s = (FileRotationPlugin *) self;
-
-
-  g_free(s->interval);
-  s->interval = g_strdup(interval);
+  self->interval = interval;
 }
 
 void

--- a/modules/file-rotation/file-rotation.h
+++ b/modules/file-rotation/file-rotation.h
@@ -29,13 +29,7 @@
 #include "signal-slot-connector/signal-slot-connector.h"
 
 typedef struct _FileRotationPlugin FileRotationPlugin;
-typedef struct _LogDriver LogDriver;
 typedef struct _FileRotationResponseData FileRotationResponseData;
-typedef enum
-{
-  FILE_ROTATION_INTERVAL,
-  FILE_ROTATION_SIZE,
-} FileRotationOption;
 
 typedef enum
 {
@@ -43,20 +37,13 @@ typedef enum
   FILE_ROTATION_FAILURE,
 } FileRotationResult;
 
-struct _FileFlushSignalData
-{
-  GString *filename;
-  gchar *interval;
-  gsize size;
-};
-
 struct _FileRotationPlugin
 {
   LogDriverPlugin super;
   SignalSlotConnector *ssc;
   gchar *interval;
   gsize size;
-  GString *filename;
+  gchar *filename;
 };
 
 struct _FileRotationResponseData
@@ -68,13 +55,9 @@ struct _FileRotationResponseData
   gint64 last_interval_check;
 };
 
-
-
-#define signal_file_rotation_response SIGNAL(file_rotation, rotation_response, FileRotationResponseData *)
-
 FileRotationPlugin *file_rotation_new(void);
 
 void file_rotation_set_size(FileRotationPlugin *self, gsize size);
-void log_rotation_set_interval(FileRotationPlugin *self, gchar *interval);
+void file_rotation_set_interval(FileRotationPlugin *self, gchar *interval);
 
 #endif

--- a/modules/file-rotation/tests/test_file_rotation.c
+++ b/modules/file-rotation/tests/test_file_rotation.c
@@ -30,8 +30,9 @@ Test(file_rotation, test_example)
 {
   FileRotationPlugin *fr = file_rotation_new();
   file_rotation_set_size(fr, 100);
-  log_rotation_set_interval(fr, "daily");
+  file_rotation_set_interval(fr, "daily");
   log_driver_plugin_free((LogDriverPlugin *) fr);
 
-  cr_assert(0);
+  cr_assert_eq(fr->size, 100);
+  cr_assert_str_eq(fr->interval, "daily");
 }

--- a/tests/copyright/policy
+++ b/tests/copyright/policy
@@ -115,6 +115,7 @@ libtest
 syslog-ng(-ctl)?
 modules/java-modules/common
 modules/java/(native|proxies|src)
+modules/affile/file-signals.h
 modules/native
 modules/http/http-signals.h
 tests/loggen

